src/Firestore/DocumentReference.php

```diff
@@ -0,0 +1,444 @@
+     *
+     * Unless `$options.merge` is set to true, this method will replace all
+     * existing document data.
+     *
+     * Example:
+     * ```
+     * $document->set([
+     *     'name' => 'Dave'
+     * ]);
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @see https://cloud.google.com/firestore/docs/reference/rpc/google.firestore.v1beta1#google.firestore.v1beta1.Firestore.Commit Commit
+     * @codingStandardsIgnoreEnd
+     *
+     * @param array $fields An array containing fields, where keys are the field
+     *        names, and values are field values. Nested arrays are allowed.
+     *        Note that unlike {@see Google\Cloud\Firestore\DocumentReference::updatePaths()},
+     *        field paths are NOT supported by this method.
+     * @param array $options {
+     *     Configuration Options
+     *
+     *     @type bool $merge If true, unwritten fields will be preserved.
+     *           Otherwise, they will be overwritten (removed). **Defaults to**
+     *           `false`.
+     * }
+     * @codingStandardsIgnoreStart
+     * @return array [WriteResult](https://cloud.google.com/firestore/docs/reference/rpc/google.firestore.v1beta1#google.firestore.v1beta1.WriteResult)
+     * @codingStandardsIgnoreEnd
+     */
+    public function set(array $fields, array $options = [])
+    {
+        return $this->writeResult(
+            $this->batchFactory()
+                ->set($this->name, $fields, $options)
+                ->commit($options)
+        );
+    }
+
+    /**
+     * Update a Firestore document.
+     *
+     * Merges provided data with data stored in Firestore.
+     *
+     * Calling this method on a non-existent document will raise an exception.
+     *
+     * This method supports various sentinel values, to perform special operations
+     * on fields. Available sentinel values are provided as methods, found in
+     * {@see Google\Cloud\Firestore\FieldValue}.
+     *
+     * To update a document using field paths, see
+     * {@see Google\Cloud\Firestore\DocumentReference::updatePaths()}.
+     *
+     * Example:
+     * ```
+     * $document->update([
+     *     'name' => 'John',
+     *     'country' => 'USA',
+     *     'cryptoCurrencies' => [
+     *         'bitcoin' => 0.5,
+     *         'ethereum' => 10,
+     *         'litecoin' => 5.51
+     *     ]
+     * ]);
+     * ```
+     *
+     * ```
+     * // Remove a field using the {@see Gooogle\Cloud\Firestore\FieldValue::deleteField()} special value.
+     * use Google\Cloud\Firestore\FieldValue;
+     *
+     * $document->update([
+     *     'country' => FieldValue::deleteField()
+     * ]);
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @see https://cloud.google.com/firestore/docs/reference/rpc/google.firestore.v1beta1#google.firestore.v1beta1.Firestore.Commit Commit
+     *
+     * @param array $fields An array containing field names paired with their value.
```

Comment:
Not at all. Updating using field paths is supported via `updatePaths()`. Since PHP does not allow objects as array keys, we can't accept instances of FieldPath as the key in a `[key, value]` array. It didn't make sense to accept dot-delimited strings without FieldPaths, because that would make dealing with escaping problematic. Therefore, we opted to simply move all field path functionality to the method which was added to handle it.